cmd/tests/tracing_module_test.go

```diff
@@ -127,6 +128,236 @@ func TestTracingClient_DoesNotInterfereWithHTTPModule(t *testing.T) {
 	assert.Equal(t, 2, gotInstrumentedRequests)
 }
 
+func TestTracingInstrumentHTTP_W3C(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	gotRequests := 0
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		gotRequests++
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing");
+			http.get("HTTPBIN_IP_URL/tracing");
+			http.head("HTTPBIN_IP_URL/tracing");
+			http.options("HTTPBIN_IP_URL/tracing");
+			http.patch("HTTPBIN_IP_URL/tracing");
+			http.post("HTTPBIN_IP_URL/tracing");
+			http.put("HTTPBIN_IP_URL/tracing");
+			http.request("GET", "HTTPBIN_IP_URL/tracing");
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, 8, gotRequests)
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	gotHTTPDataPoints := false
+
+	for _, jsonLine := range bytes.Split(jsonResults, []byte("\n")) {
+		if len(jsonLine) == 0 {
+			continue
+		}
+
+		var line sampleEnvelope
+		require.NoError(t, json.Unmarshal(jsonLine, &line))
+
+		if line.Type != "Point" {
+			continue
+		}
+
+		// Filter metric samples which are not related to http
+		if !strings.HasPrefix(line.Metric, "http_") {
+			continue
+		}
+
+		gotHTTPDataPoints = true
+
+		anyTraceID, hasTraceID := line.Data.Metadata["trace_id"]
+		require.True(t, hasTraceID)
+
+		traceID, gotTraceID := anyTraceID.(string)
+		require.True(t, gotTraceID)
+
+		assert.Len(t, traceID, 32)
+	}
+
+	assert.True(t, gotHTTPDataPoints)
+}
+
+func TestTracingInstrumentHTTP_Jaeger(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	gotRequests := 0
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		gotRequests++
+		assert.NotEmpty(t, r.Header.Get(tracing.JaegerHeaderName))
+		assert.Len(t, r.Header.Get(tracing.JaegerHeaderName), 45)
```

Comment:
I would argue those are likely better as just inline constants instead of importing the whole package(not that you don't indirectly). This also means that if you somehow change it in the tracing API it won't suddenly break while the test is still passing ;)